File: npm_api/examples/subscription.ts
// Follow a stream's lifecycle and recording events as they happen.
import { TenantEventsDocument } from "@livepeer-frameworks/api";
import { createSubscriptionClient } from "@livepeer-frameworks/api/subscriptions";

const subscriptions = createSubscriptionClient({
  url: "wss://bridge.example.com/graphql/ws",
  // Read again for every connection, so a refreshed token is used after a reconnect.
  token: () => process.env.FRAMEWORKS_API_TOKEN,
});

for await (const { tenantEvents: event } of subscriptions.subscribe(TenantEventsDocument, {
  types: ["stream.live", "stream.idle", "recording.ready"],
})) {
  switch (event.data.__typename) {
    case "StreamLive":
      console.log(`${event.data.streamId} went live at ${event.time}`);
      break;
    case "RecordingReady":
      console.log(`recording ${event.data.artifact?.artifactId} is ready`);
      break;
    default:
      console.log(event.type);
  }
}

File: sdk_python/examples/subscription.py
"""Follow the live and idle events of every stream."""

import asyncio
import os

from livepeer_frameworks import AsyncFrameWorksClient, AuthenticationError


async def main() -> None:
    async with AsyncFrameWorksClient(
        "https://bridge.frameworks.network/graphql", token=os.environ["FRAMEWORKS_API_TOKEN"]
    ) as fw:
        try:
            async for message in fw.tenant_events(types=["stream.live", "stream.idle"]):
                event = message.tenant_events
                print(event.time, event.type_, event.subject)
        except AuthenticationError:
            print("the token was rejected; the subscription does not retry with it")


if __name__ == "__main__":
    asyncio.run(main())
